### PR TITLE
Add MessageProducerClientBuilder for creating configurable message pr…

### PR DESCRIPTION
**Pull request by allanmckenzie**
…oducer clients

### CODE REVIEW

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClientBuilder.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        this.maxRetryInterval = of(maxRetryInterval);
+        return this;
+    }
+
+    public MessageProducerClientBuilder withRetryIntervalMultiplier(final double retryIntervalMultiplier) {
+        this.retryIntervalMultiplier = of(retryIntervalMultiplier);
+        return this;
+    }
+
+    public MessageProducerClientBuilder withReconnectAttempts(int reconnectAttempts) {
```
Comment (Morbu) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClientBuilder.java:44]: final :)

Comment (allanmckenzie) [reply test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClientBuilder.java:44]: Noooooooooooooooooooooo

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java [hunk 2]
```diff
@@ -37,8 +48,8 @@ public class MessageProducerClient implements AutoCloseable {
     public void startProducer(final String topicName) {
 
         try {
-            final ActiveMQConnectionFactory factory = new ActiveMQConnectionFactory(QUEUE_URI);
-            connection = factory.createConnection();
+            activeMQConnectionFactory.setBrokerURL(QUEUE_URI);
```
Comment (sanintel3) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java:51]: Setting QUEUE_URI through constructor is wrapping this statement in a try catch and throwing IllegalStateException (Inside ActiveMQConnectionFactory class), not entirely sure what exactly is happening in this setBrokerURL, may be worth replicating it here as well

Comment (allanmckenzie) [reply test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java:51]: This is what is happening in setBrokerURL()

if (this.readOnly) {
            throw new javax.jms.IllegalStateException("You cannot use setBrokerURL after the connection factory has been used");
        }

Comment (allanmckenzie) [reply test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java:51]: It's to stop people starting the connection then changing the url, which would be daft. However, we've not started the connection factory yet in out code, so it's academic

**Pull request merged**